purchase_merge/wizard/purchase_merge.py

```diff
@@ -0,0 +1,218 @@
+                    [str(x) for x in picking_type_names]
+                )
+            )
+        if len(incoterm_names) > 1:
+            error_msg += "{} {}: {}\n".format(
+                start_msg, _('incoterms'), ", ".join(
+                    [str(x) for x in incoterm_names]
+                )
+            )
+        if len(payment_term_names) > 1:
+            error_msg += "{} {}: {}\n".format(
+                start_msg, _('payment terms'), ", ".join(
+                    [str(x) for x in payment_term_names]
+                )
+            )
+        if len(fiscal_position_names) > 1:
+            error_msg += "{} {}: {}\n".format(
+                start_msg, _('fiscal positions'), ", ".join(
+                    [str(x) for x in fiscal_position_names]
+                )
+            )
+
+        if error_msg:
+            raise UserError(error_msg)
+
+    # ----------------------------------------
+    # Update method
+    # ----------------------------------------
+
+    @api.model
+    def _update_values(self, src_purchase, dst_purchase):
+        """ Update values of dst_purchase with the ones from the src_purchase.
+            :param src_purchase : recordset of source purchase.order
+            :param dst_purchase : record of destination purchase.order
+        """
+        # initialize destination origin and partner_ref
+        origin = set([dst_purchase.origin or ""])
+        origin.update(set([x.origin for x in src_purchase if x.origin]))
+
+        partner_ref = set([dst_purchase.partner_ref or ""])
+        partner_ref.update(
+            set([x.partner_ref for x in src_purchase if x.partner_ref])
+        )
+
+        # update destination origin and partner_ref
+        po_names = src_purchase.mapped('name')
+        src_order_line = src_purchase.mapped('order_line')
+        # merge all order lines + set origin and partner_ref
+        dst_purchase.write({
+            'order_line': [(4, line, 0) for line in src_order_line.ids],
+            'origin': ', '.join(origin),
+            'partner_ref': ', '.join(partner_ref),
+        })
+        for po in src_purchase:
+            self._add_message('to', [dst_purchase.name], po)
+        self._add_message('from', po_names, dst_purchase)
+
+    def _add_message(self, way, po_name, po):
+        """ Send a message post with to advise the po about the merge.
+            :param way : choice between 'from' or 'to'
+            :param po_name : list of purchase order name to add in the body
+            :param po_name : the po where the message will be posted
+        """
+        subject = 'Merge purchase order'
+        body = _('This purchase order lines have been merged {} : {}'.format(
+            way, ' ,'.join(po_name)
+        ))
+
+        po.message_post(
+            body=body,
+            subject=subject,
+            content_subtype='plaintext'
+        )
+
+    def _merge(self, purchases, dst_purchase=None):
```

Comment:
add a comment saying that it could be improved using https://github.com/OCA/openupgradelib/blob/master/openupgradelib/openupgrade_merge_records.py
as Pedro said.
And add this to the ROADMAP.